src/main/java/uk/gov/hmcts/cmc/claimstore/rules/PaidInFullRule.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.cmc.claimstore.rules;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.exceptions.ClaimantLinkException;
+import uk.gov.hmcts.cmc.claimstore.exceptions.ConflictException;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+@Component
+public class PaidInFullRule {
+
+    public void assertPaidInFull(Claim claim, String claimantId) {
+        requireNonNull(claim, "claim object can not be null");
+        requireNonNull(claimantId, "submitterId object can not be null");
+        if (!claim.getSubmitterId().equals(claimantId)) {
```

Comment:
Suggest flipping `.equals` as this unit knows `claimantId` is not-null at this point, but doesn't know about `claim.getSubmitterId()`

Reply:
can not imagine a claim object without a submitter id.